.github/workflows/build.yml

```diff
@@ -29,6 +29,12 @@ jobs:
     - name: Build base image
       run: sudo ./build.sh
 
+    - name: Get base image version info
+      id: version
+      run: |
+        docker run ghcr.io/steamdeckhomebrew/holo-base:latest \
```

Comment:
I'd prefer if we instead get it from the updater API - perhaps download.sh could be split out of the build.sh step or something

Reply:
Interesting, I thought `/etc/os-release` would be more reliable than the API, since most of the time I was just getting the string `"snapshot"` back as the "version" from the API. Also, presumably it would be a good way for people to verify their host's version vs the container version 

Based on your comment https://github.com/SteamDeckHomebrew/holo-docker/issues/6#issuecomment-1731735835 though, maybe I can find a more accurate version number using the API and spit it out after the download. At minimum we should be able to get the `buildid` that way.

Reply:
Circling back on this, I wrote a little script to crawl all of https://steamdeck-atomupd.steamos.cloud/meta/steamos/amd64 in an attempt to find where the 3.4 packages live, and I couldn't find any JSON files at all with `"version": "3.4.x"` — only `"3.5.0"`, `"3.6.0"`, and `"snapshot"`. Do you know of some way to get the numerical version for stable releases, rather than just preview and beta versions? So far nothing I have tried on the `/updates` API has worked either.

Reply:
Ok, one more follow-up. I finally found the actual code that is used to look for updates, and tried running it against both my stable partition and my 3.5 partition to see what the difference in output is.
<details>
<summary>Stable manifest file</summary>

```
$ sudo steamos-atomupd-client -d --query-only --manifest-file /mnt/rootfs-A/etc/steamos-atomupd/manifest.json
DEBUG:client.py:732: Parsing config from file: /etc/steamos-atomupd/client.conf
DEBUG:client.py:566: The attempts log is missing, assuming no previous failed update attempts
DEBUG:client.py:800: Using manifest file '/mnt/rootfs-A/etc/steamos-atomupd/manifest.json'
INFO:client.py:831: MetaURL is either missing or not working, falling back to QueryURL
DEBUG:client.py:335: Downloading update file https://steamdeck-atomupd.steamos.cloud/updates?product=steamos&release=holo&variant=steamdeck&arch=amd64&version=snapshot&buildid=20230508.1&checkpoint=False&estimated_size=0
INFO:client.py:839: Server returned something, guess an update is available
DEBUG:client.py:844: Parsing update file: /run/steamos-atomupd/update.json
DEBUG:client.py:867: An update is available for release 'holo'
DEBUG:client.py:872: > going to version: None (20230831.1)
DEBUG:client.py:873: > update path: steamdeck/20230831.1/steamdeck-20230831.1-snapshot.raucb
{
  "minor": {
    "release": "holo",
    "candidates": [
      {
        "image": {
          "product": "steamos",
          "release": "holo",
          "variant": "steamdeck",
          "arch": "amd64",
          "version": "snapshot",
          "buildid": "20230831.1",
          "checkpoint": false,
          "estimated_size": 0
        },
        "update_path": "steamdeck/20230831.1/steamdeck-20230831.1-snapshot.raucb"
      }
    ]
  }
}
```


</details>

As you can see, it just returns `version="snapshot"` which is specifically treated as  "none" in the code (`/usr/lib/python3.10/site-packages/steamosatomupd/image.py:Image.from_values`):
```py
        # Parse version, raise ValueError if need be
        if version_str == 'snapshot':
            version = None
```


<details>
<summary>3.5 (beta candidate) manifest</summary>

```
$ sudo steamos-atomupd-client -d --query-only
DEBUG:client.py:732: Parsing config from file: /etc/steamos-atomupd/client.conf
DEBUG:client.py:566: The attempts log is missing, assuming no previous failed update attempts
DEBUG:client.py:800: Using manifest file '/etc/steamos-atomupd/manifest.json'
INFO:client.py:831: MetaURL is either missing or not working, falling back to QueryURL
DEBUG:client.py:335: Downloading update file https://steamdeck-atomupd.steamos.cloud/updates?product=steamos&release=holo&variant=steamdeck-bc&arch=amd64&version=3.5.0&buildid=20230915.100&checkpoint=False&estimated_size=0
INFO:client.py:839: Server returned something, guess an update is available
DEBUG:client.py:844: Parsing update file: /run/steamos-atomupd/update.json
DEBUG:client.py:851: We are up to date
```

</details>



So, I think for stable patch updates we won't be able to get the version info from the API and `/etc/os-release` might be the best option. Actually, the updater code also has fallback logic to read from `/etc/os-release` when the `manifest.json` isn't found (`/usr/lib/python3.10/site-packages/steamosatomupd/client.py:UpdateClient.run`), so I think it probably will be up-to-date normally.

Alternatively we could read from the `manifest.json` itself, but then we will most of the time end up with a version of `snapshot`, which may not be very helpful for people trying to figure out which patch version their container is using (of course `buildid` is the most specific).

Reply:
but we *already* get the version from the API. it's part of the download URL that download.sh gets. look at that response

Reply:
Sorry, my notes above were a little rambling and I forgot to make my main point: the `"version"` the API returns is not actually always a numerical version. Example:
```console
$ curl -sS --user $AUTH "https://steamdeck-atomupd.steamos.cloud/updates?product=steamos&release=holo&variant=steamdeck&arch=amd64&version=snapshot&buildid=20220526.1&checkpoint=False&estimated_size=0" | jq
{
  "minor": {
    "release": "holo",
    "candidates": [
      {
        "image": {
          "product": "steamos",
          "release": "holo",
          "variant": "steamdeck",
          "arch": "amd64",
          "version": "snapshot",
          "buildid": "20230831.1",
          "checkpoint": false,
          "estimated_size": 0
        },
        "update_path": "steamdeck/20230831.1/steamdeck-20230831.1-snapshot.raucb"
      }
    ]
  }
}
```

I don't think tagging something as `snapshot` is likely to be helpful for most consumers of these images.

Interestingly, the beta variant does seem to have a version, so I'm not sure what determines whether it's set or not (I would have expected the opposite, if anything). This is exactly the same request but with `variant=steamdeck-beta`:
```console
$ curl -sS --user $AUTH "https://steamdeck-atomupd.steamos.cloud/updates?product=steamos&release=holo&variant=steamdeck-beta&arch=amd64&version=snapshot&buildid=20220526.1&checkpoint=False&estimated_size=0" | jq
{
  "minor": {
    "release": "holo",
    "candidates": [
      {
        "image": {
          "product": "steamos",
          "release": "holo",
          "variant": "steamdeck-beta",
          "arch": "amd64",
          "version": "3.5.0",
          "buildid": "20230915.100",
          "checkpoint": false,
          "estimated_size": 0
        },
        "update_path": "steamdeck/20230915.100/steamdeck-20230915.100-3.5.0.raucb"
      }
    ]
  }
}
```